slackevents/inner_events.go

```diff
@@ -559,61 +558,110 @@ type TeamDomainChangeEvent struct {
+	Type            string              `json:"type"`
+	Invite          *SharedInvite       `json:"invite"`
+	Channel         *slack.Conversation `json:"channel"`
+	ApprovingTeamID string              `json:"approving_team_id"`
+	TeamsInChannel  []*SlackEventTeam   `json:"teams_in_channel"`
+	ApprovingUser   *SlackEventUser     `json:"approving_user"`
+	EventTs         string              `json:"event_ts"`
 }
 
+// SharedChannelInviteAcceptedEvent is sent if external org accepts a Slack Connect channel invite
 type SharedChannelInviteAcceptedEvent struct {
-	Type             string         `json:"type"`
-	ApprovalRequired bool           `json:"approval_required"`
-	TeamsInChannel   []*slack.Team  `json:"teams_in_channel"`
-	EventTs          string         `json:"event_ts"`
-	Channel          *slack.Channel `jsopn:"channel"`
-	AcceptingUser    *slack.User    `json:"accepting_user"`
-	Invite           *Invite        `json:"invite"`
+	Type                string            `json:"type"`
+	ApprovalRequired    bool              `json:"approval_required"`
+	Invite              *SharedInvite     `json:"invite"`
+	Channel             *SharedChannel    `json:"channel"`
+	TeamsInChannel      []*SlackEventTeam `json:"teams_in_channel"`
+	AcceptingUser       *SlackEventUser   `json:"accepting_user"`
+	EventTs             string            `json:"event_ts"`
+	RequiresSponsorship bool              `json:"requires_sponsorship,omitempty"`
 }
 
-type SharedChannelInviteApprovedEvent struct {
-	Type            string         `json:"type"`
-	Invite          *Invite        `json:"invite"`
-	Channel         *slack.Channel `jsopn:"channel"`
-	ApprovingTeamID string         `json:"approving_team_id"`
-	TeamsInChannel  []*slack.Team  `json:"teams_in_channel"`
-	ApprovingUser   *slack.User    `json:"approving_user"`
-	EventTs         string         `json:"event_ts"`
+// SharedChannelInviteDeclinedEvent is sent if external or internal org declines the Slack Connect invite
+type SharedChannelInviteDeclinedEvent struct {
+	Type            string            `json:"type"`
+	Invite          *SharedInvite     `json:"invite"`
+	Channel         *SharedChannel    `json:"channel"`
+	DecliningTeamID string            `json:"declining_team_id"`
+	TeamsInChannel  []*SlackEventTeam `json:"teams_in_channel"`
+	DecliningUser   *SlackEventUser   `json:"declining_user"`
+	EventTs         string            `json:"event_ts"`
 }
 
-type SharedChannelInviteDeclinedEvent struct {
-	Type            string         `json:"type"`
-	Invite          *Invite        `json:"invite"`
-	Channel         *slack.Channel `jsopn:"channel"`
-	DecliningTeamID string         `json:"declining_team_id"`
-	DecliningUser   *slack.User    `json:"declining_user"`
-	TeamsInChannel  []*slack.Team  `json:"teams_in_channel"`
-	EventTs         string         `json:"event_ts"`
+// SharedChannelInviteReceivedEvent is sent if a bot or app is invited to a Slack Connect channel
+type SharedChannelInviteReceivedEvent struct {
```

Comment:
A bunch of the stuff you had ported over got upstreamed, so our diff to the main repo is actually much smaller now.